### ISSUE TITLE
Cannot build from source on Ubuntu 22.04.1 LTS (Google Cloud)

### ISSUE THREAD

**Issue by silvasean** [google/jax#14822]
### Description

I am trying to build JAX from source, and I'm getting stuck on this `@local_config_python//:python_include: Genrules without outputs don't make sense` error.

```
ERROR: /home/silvasean/.cache/bazel/_bazel_silvasean/2be0f603d22f00664ced77380d946e12/external/local_config_python/BUILD:78:8: in outs attribute of genrule rule @local_config_python//:python_include: Genrules without outputs don't make sense
ERROR: /home/silvasean/.cache/bazel/_bazel_silvasean/2be0f603d22f00664ced77380d946e12/external/local_config_python/BUILD:78:8: Analysis of target '@local_config_python//:python_include' failed
```

It is talking about this genrule, which seems fishy
(locally for me it is in /home/silvasean/.cache/bazel/_bazel_silvasean/2be0f603d22f00664ced77380d946e12/external/local_config_python/BUILD)

```
genrule(
    name = "python_include",
    outs = [

    ],
    cmd = """

   """,
)
```

My bazel-fu is not strong enough to really debug this. It seems to be possibly coming from tensorflow/third_party/py/python_configure.bzl but I can't figure out reproduce the build step that generates the bogus genrule to debug the issue. Any help would be appreciated.

Reproduction steps (following https://jax.readthedocs.io/en/latest/developer.html exactly as far as I can tell):
https://gist.github.com/silvasean/5ea1c43be9e2dc1d2317d985eb696a76

### What jax/jaxlib version are you using?

88d5a4110b9f902dc57ac35878c04c1eb3bf54c1

### Which accelerator(s) are you using?

GPU

### Additional system info

Ubuntu 22.04.1 LTS

### NVIDIA GPU info

```
(jax_repro.venv) silvasean@silvasean-a100-cuda117:~/jax_repro/jax$ python --version
Python 3.10.6
(jax_repro.venv) silvasean@silvasean-a100-cuda117:~/jax_repro$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.1 LTS"
(jax_repro.venv) silvasean@silvasean-a100-cuda117:~/jax_repro$ nvidia-smi
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 515.43.04    Driver Version: 515.43.04    CUDA Version: 11.7     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA A100-SXM...  On   | 00000000:00:04.0 Off |                    0 |
| N/A   30C    P0    46W / 400W |      0MiB / 40960MiB |      0%      Default |
|                               |                      |             Disabled |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```

**Comment by jpienaar** [comment]
Did you also try https://github.com/openxla/openxla-pjrt-plugin#building-jax-from-source ?

**Comment by silvasean** [comment]
> Did you also try https://github.com/openxla/openxla-pjrt-plugin#building-jax-from-source ?

Yes, that's actually where I started. I then found that the vanilla jax build doesn't work.

**Comment by GMNGeoffrey** [comment]
I think it's coming from here: https://github.com/tensorflow/tensorflow/blob/master/third_party/py/python_configure.bzl

I can help debug further later today if you need it

**Comment by hawkinsp** [comment]
I can't reproduce this. I took a fresh Ubuntu 22.04.2 VM, ran:

```
sudo apt install python3-pip
pip install numpy
git clone https://github.com/google/jax.git
cd jax
python build/build.py
```
and everything built fine.

You'll have to share more detailed reproduction instructions.

**Comment by silvasean** [comment]
Thanks @hawkinsp, that narrowed this down. The issue is related to having `python3-dev` installed, and my system got into a "bad state" where bazel had the "not python3-dev installed" state cached. It appears that cache invalidation retains its position in the pantheon of most difficult problems in computer science :rofl: 

To explain how I got here:

1. I followed the steps in openxla-pjrt-plugin for building JAX from source: [link](https://github.com/openxla/openxla-pjrt-plugin#building-jax-from-source)
2. The steps there don't include installing python3-dev
3. I did the bazel build, and it cached the "not having python3-dev installed" state, which created the bogus genrule
4. Later when I installed python3-dev, it was not seen and the issue persisted.

The solution was ensuring python3-dev was installed and removing `~/.cache/bazel`.

Also, an interesting side note: In your instructions you install `apt install python3-pip` which installs `python3-dev` incidentally. In my flow I got pip via `apt install python3-venv` which installs the `python3-pip-whl` Ubuntu package as a byproduct. But `python3-pip-whl` does not install `python3-dev`, unlike `python3-pip`.

Phew
